File: wiringPi/c++/src/digital_input.cpp
//
// Program: digital_input.cpp
// Purpose: Reads a digital input and displays its value
//          using wiringPi library
// Author:  Aldo Nunez
//

#include <iostream>
#include <wiringPi.h>

using namespace std;

int
main ( void )
{
    const int PORT = 7;

    if ( wiringPiSetup () )
    {
        cerr <<  "Failing to setup wiringPi" << endl;
        return 1;
    }

    // sets the port as input
    pinMode ( PORT, INPUT );

    // reads input value
    int value = digitalRead ( PORT );

    cout << "Input port " <<  PORT << ": " << value << endl;

    return 0;
}

File: wiringPi/python/digital_input.py
#!/usr/bin/env python

'''
Program: blink.py
Purpose: reads a digital input and 
         displays its value using
         wiringPi library
Author:  Aldo Nunez
'''

# Import GPIO module
import wiringpi as wipi

def main ():
    # setup wiringpi
    wipi.wiringPiSetup ()

    # wPi 7 corresponds to Pin 7
    WPI_PORT = 7;


    # '0' means input
    wipi.pinMode ( WPI_PORT, 0 )

    value = wipi.digitalRead ( WPI_PORT )
    print ( 'Input port {}: {}'.format ( WPI_PORT, value ) )

if __name__  == "__main__":
    main ()
